### PR TITLE
Adding a new feature on the 3D in single and multi coil MRI reconstruction

### PR DESCRIPTION
**Pull request by LElgueddari**
Adding a new folder in the plugins/mri to add the 3D reconstruction methods for both single and multi coil reconstruction method
Adding a dependency on pyWavelet to have 3D Wavelet transform
Adding 4 new exemples for the 3D case with FFT/ NFFT songle/multiple coils
Changing a little bit the parallel mri gradient to make it compatible with the 3D setting
Changing the formating of the sensitivity maps for both 2D and 3D to generalize a bit the code
Adding some test on the fourier operations and its adjoint
Adding some test on the optimizers
Adding the data in the loader file
Creating a new loader for the .mat file

### CODE REVIEW

File: pysap/plugins/mri/reconstruct/fourier.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def op(self, img):
             masked Fourier transform of the input image.
         """
         self.plan.f_hat = img
-        return self.plan.trafo()
+        return (1.0 / np.sqrt(1.0*self.plan.M)) * self.plan.trafo()
```
[the remark this answers is not in the document]
Comment (LElgueddari) [reply pysap/plugins/mri/reconstruct/fourier.py]: This is only to have a symmetric fourier transform, it's not a bug

File: pysap/plugins/mri/reconstruct/fourier.py [hunk 2]
```diff
@@ -178,4 +178,4 @@ def adj_op(self, x):
             inverse 2D discrete Fourier transform of the input coefficients.
         """
         self.plan.f = x
-        return (1.0 / self.plan.M) * self.plan.adjoint()
+        return (1.0 / np.sqrt(self.plan.M)) * self.plan.adjoint()
```
[the remark this answers is not in the document]
Comment (LElgueddari) [reply pysap/plugins/mri/reconstruct/fourier.py]: Same as previous

File: pysap/plugins/mri/reconstruct/linear.py [hunk 3]
```diff
@@ -45,6 +47,12 @@ def __init__(self, wavelet_name, nb_scale=4, verbose=0):
             nb_scale=self.nb_scale, verbose=verbose)
         self.coeffs_shape = None
 
+    def get_coeff(self):
+        return self.transform.analysis_data
+
+    def set_coeff(self, coeffs):
+        self.transform.analysis_data = coeffs
+
```
[the remark this answers is not in the document]
Comment (LElgueddari) [reply pysap/plugins/mri/reconstruct/linear.py]: It's not a new feature
It is suppose to make the code compatible with the pywavelet package and sparse2D/3D

**Comment by AGrigis** [comment]
MRI plugins is now here: https://github.com/CEA-COSMIC/pysap-mri.git
Please resubmit your PR.

**Pull request closed without merging**